### ISSUE TITLE
Cannot load class graphQLServletRegistrationBean. 

### ISSUE THREAD

**Issue by gurgl** [graphql-java-kickstart/graphql-java-tools#252]
Running "example" master from IntelliJ with java 8 i get

Cannot load class graphQLServletRegistrationBean. Error creating bean with name 'graphQLServletRegistrationBean': Requested bean is currently in creation: Is there an unresolvable circular reference?

or full log : 

```
/usr/lib/jvm/java-8-oracle/bin/java -XX:TieredStopAtLevel=1 -noverify -Dspring.output.ansi.enabled=always -Dcom.sun.management.jmxremote -Dspring.liveBeansView.mbeanDomain -Dspring.application.admin.enabled=true -javaagent:/usr/local/bin/idea-IU-183.4284.148/lib/idea_rt.jar=33223:/usr/local/bin/idea-IU-183.4284.148/bin -Dfile.encoding=UTF-8 -classpath /usr/lib/jvm/java-8-oracle/jre/lib/charsets.jar:/usr/lib/jvm/java-8-oracle/jre/lib/deploy.jar:/usr/lib/jvm/java-8-oracle/jre/lib/ext/cldrdata.jar:/usr/lib/jvm/java-8-oracle/jre/lib/ext/dnsns.jar:/usr/lib/jvm/java-8-oracle/jre/lib/ext/jaccess.jar:/usr/lib/jvm/java-8-oracle/jre/lib/ext/jfxrt.jar:/usr/lib/jvm/java-8-oracle/jre/lib/ext/localedata.jar:/usr/lib/jvm/java-8-oracle/jre/lib/ext/nashorn.jar:/usr/lib/jvm/java-8-oracle/jre/lib/ext/sunec.jar:/usr/lib/jvm/java-8-oracle/jre/lib/ext/sunjce_provider.jar:/usr/lib/jvm/java-8-oracle/jre/lib/ext/sunpkcs11.jar:/usr/lib/jvm/java-8-oracle/jre/lib/ext/zipfs.jar:/usr/lib/jvm/java-8-oracle/jre/lib/javaws.jar:/usr/lib/jvm/java-8-oracle/jre/lib/jce.jar:/usr/lib/jvm/java-8-oracle/jre/lib/jfr.jar:/usr/lib/jvm/java-8-oracle/jre/lib/jfxswt.jar:/usr/lib/jvm/java-8-oracle/jre/lib/jsse.jar:/usr/lib/jvm/java-8-oracle/jre/lib/management-agent.jar:/usr/lib/jvm/java-8-oracle/jre/lib/plugin.jar:/usr/lib/jvm/java-8-oracle/jre/lib/resources.jar:/usr/lib/jvm/java-8-oracle/jre/lib/rt.jar:/home/someuser/Documents/src/3rdParty/graphql-java-tools/example/target/classes:/home/someuser/.m2/repository/org/springframework/boot/spring-boot-starter/2.1.0.RELEASE/spring-boot-starter-2.1.0.RELEASE.jar:/home/someuser/.m2/repository/org/springframework/boot/spring-boot/2.1.0.RELEASE/spring-boot-2.1.0.RELEASE.jar:/home/someuser/.m2/repository/org/springframework/spring-context/5.1.2.RELEASE/spring-context-5.1.2.RELEASE.jar:/home/someuser/.m2/repository/org/springframework/boot/spring-boot-autoconfigure/2.1.0.RELEASE/spring-boot-autoconfigure-2.1.0.RELEASE.jar:/home/someuser/.m2/repository/org/springframework/boot/spring-boot-starter-logging/2.1.0.RELEASE/spring-boot-starter-logging-2.1.0.RELEASE.jar:/home/someuser/.m2/repository/ch/qos/logback/logback-classic/1.2.3/logback-classic-1.2.3.jar:/home/someuser/.m2/repository/ch/qos/logback/logback-core/1.2.3/logback-core-1.2.3.jar:/home/someuser/.m2/repository/org/apache/logging/log4j/log4j-to-slf4j/2.11.1/log4j-to-slf4j-2.11.1.jar:/home/someuser/.m2/repository/org/apache/logging/log4j/log4j-api/2.11.1/log4j-api-2.11.1.jar:/home/someuser/.m2/repository/org/slf4j/jul-to-slf4j/1.7.25/jul-to-slf4j-1.7.25.jar:/home/someuser/.m2/repository/javax/annotation/javax.annotation-api/1.3.2/javax.annotation-api-1.3.2.jar:/home/someuser/.m2/repository/org/springframework/spring-core/5.1.2.RELEASE/spring-core-5.1.2.RELEASE.jar:/home/someuser/.m2/repository/org/springframework/spring-jcl/5.1.2.RELEASE/spring-jcl-5.1.2.RELEASE.jar:/home/someuser/.m2/repository/org/yaml/snakeyaml/1.23/snakeyaml-1.23.jar:/home/someuser/.m2/repository/org/springframework/boot/spring-boot-starter-web/2.1.0.RELEASE/spring-boot-starter-web-2.1.0.RELEASE.jar:/home/someuser/.m2/repository/org/springframework/boot/spring-boot-starter-json/2.1.0.RELEASE/spring-boot-starter-json-2.1.0.RELEASE.jar:/home/someuser/.m2/repository/com/fasterxml/jackson/core/jackson-databind/2.9.7/jackson-databind-2.9.7.jar:/home/someuser/.m2/repository/com/fasterxml/jackson/datatype/jackson-datatype-jsr310/2.9.7/jackson-datatype-jsr310-2.9.7.jar:/home/someuser/.m2/repository/com/fasterxml/jackson/module/jackson-module-parameter-names/2.9.7/jackson-module-parameter-names-2.9.7.jar:/home/someuser/.m2/repository/org/hibernate/validator/hibernate-validator/6.0.13.Final/hibernate-validator-6.0.13.Final.jar:/home/someuser/.m2/repository/javax/validation/validation-api/2.0.1.Final/validation-api-2.0.1.Final.jar:/home/someuser/.m2/repository/org/jboss/logging/jboss-logging/3.3.2.Final/jboss-logging-3.3.2.Final.jar:/home/someuser/.m2/repository/org/springframework/spring-web/5.1.2.RELEASE/spring-web-5.1.2.RELEASE.jar:/home/someuser/.m2/repository/org/springframework/spring-beans/5.1.2.RELEASE/spring-beans-5.1.2.RELEASE.jar:/home/someuser/.m2/repository/org/springframework/spring-webmvc/5.1.2.RELEASE/spring-webmvc-5.1.2.RELEASE.jar:/home/someuser/.m2/repository/org/springframework/spring-aop/5.1.2.RELEASE/spring-aop-5.1.2.RELEASE.jar:/home/someuser/.m2/repository/org/springframework/spring-expression/5.1.2.RELEASE/spring-expression-5.1.2.RELEASE.jar:/home/someuser/.m2/repository/org/springframework/boot/spring-boot-starter-undertow/2.1.0.RELEASE/spring-boot-starter-undertow-2.1.0.RELEASE.jar:/home/someuser/.m2/repository/io/undertow/undertow-core/2.0.14.Final/undertow-core-2.0.14.Final.jar:/home/someuser/.m2/repository/org/jboss/xnio/xnio-api/3.3.8.Final/xnio-api-3.3.8.Final.jar:/home/someuser/.m2/repository/org/jboss/xnio/xnio-nio/3.3.8.Final/xnio-nio-3.3.8.Final.jar:/home/someuser/.m2/repository/io/undertow/undertow-servlet/2.0.14.Final/undertow-servlet-2.0.14.Final.jar:/home/someuser/.m2/repository/org/jboss/spec/javax/annotation/jboss-annotations-api_1.2_spec/1.0.2.Final/jboss-annotations-api_1.2_spec-1.0.2.Final.jar:/home/someuser/.m2/repository/io/undertow/undertow-websockets-jsr/2.0.14.Final/undertow-websockets-jsr-2.0.14.Final.jar:/home/someuser/.m2/repository/org/jboss/spec/javax/websocket/jboss-websocket-api_1.1_spec/1.1.3.Final/jboss-websocket-api_1.1_spec-1.1.3.Final.jar:/home/someuser/.m2/repository/javax/servlet/javax.servlet-api/4.0.1/javax.servlet-api-4.0.1.jar:/home/someuser/.m2/repository/org/glassfish/javax.el/3.0.0/javax.el-3.0.0.jar:/home/someuser/.m2/repository/com/graphql-java-kickstart/graphql-spring-boot-starter/5.5.0/graphql-spring-boot-starter-5.5.0.jar:/home/someuser/.m2/repository/com/graphql-java-kickstart/graphql-spring-boot-autoconfigure/5.5.0/graphql-spring-boot-autoconfigure-5.5.0.jar:/home/someuser/.m2/repository/org/springframework/boot/spring-boot-starter-websocket/2.1.0.RELEASE/spring-boot-starter-websocket-2.1.0.RELEASE.jar:/home/someuser/.m2/repository/org/springframework/spring-messaging/5.1.2.RELEASE/spring-messaging-5.1.2.RELEASE.jar:/home/someuser/.m2/repository/org/springframework/spring-websocket/5.1.2.RELEASE/spring-websocket-5.1.2.RELEASE.jar:/home/someuser/.m2/repository/org/springframework/boot/spring-boot-starter-actuator/2.1.0.RELEASE/spring-boot-starter-actuator-2.1.0.RELEASE.jar:/home/someuser/.m2/repository/org/springframework/boot/spring-boot-actuator-autoconfigure/2.1.0.RELEASE/spring-boot-actuator-autoconfigure-2.1.0.RELEASE.jar:/home/someuser/.m2/repository/org/springframework/boot/spring-boot-actuator/2.1.0.RELEASE/spring-boot-actuator-2.1.0.RELEASE.jar:/home/someuser/.m2/repository/io/micrometer/micrometer-core/1.1.0/micrometer-core-1.1.0.jar:/home/someuser/.m2/repository/org/hdrhistogram/HdrHistogram/2.1.9/HdrHistogram-2.1.9.jar:/home/someuser/.m2/repository/org/latencyutils/LatencyUtils/2.0.3/LatencyUtils-2.0.3.jar:/home/someuser/.m2/repository/com/graphql-java-kickstart/graphql-java-servlet/7.2.0/graphql-java-servlet-7.2.0.jar:/home/someuser/.m2/repository/javax/websocket/javax.websocket-api/1.1/javax.websocket-api-1.1.jar:/home/someuser/.m2/repository/commons-io/commons-io/2.6/commons-io-2.6.jar:/home/someuser/.m2/repository/com/graphql-java-kickstart/graphiql-spring-boot-starter/5.5.0/graphiql-spring-boot-starter-5.5.0.jar:/home/someuser/.m2/repository/com/graphql-java-kickstart/graphiql-spring-boot-autoconfigure/5.5.0/graphiql-spring-boot-autoconfigure-5.5.0.jar:/home/someuser/.m2/repository/org/apache/commons/commons-text/1.1/commons-text-1.1.jar:/home/someuser/.m2/repository/com/graphql-java-kickstart/voyager-spring-boot-starter/5.5.0/voyager-spring-boot-starter-5.5.0.jar:/home/someuser/.m2/repository/com/graphql-java-kickstart/voyager-spring-boot-autoconfigure/5.5.0/voyager-spring-boot-autoconfigure-5.5.0.jar:/home/someuser/Documents/src/3rdParty/graphql-java-tools/target/classes:/home/someuser/.m2/repository/org/jetbrains/kotlin/kotlin-stdlib/1.3.10/kotlin-stdlib-1.3.10.jar:/home/someuser/.m2/repository/org/jetbrains/kotlin/kotlin-stdlib-common/1.3.10/kotlin-stdlib-common-1.3.10.jar:/home/someuser/.m2/repository/org/jetbrains/annotations/13.0/annotations-13.0.jar:/home/someuser/.m2/repository/org/jetbrains/kotlin/kotlin-reflect/1.3.10/kotlin-reflect-1.3.10.jar:/home/someuser/.m2/repository/org/jetbrains/kotlinx/kotlinx-coroutines-jdk8/1.0.1/kotlinx-coroutines-jdk8-1.0.1.jar:/home/someuser/.m2/repository/org/jetbrains/kotlinx/kotlinx-coroutines-core/1.0.1/kotlinx-coroutines-core-1.0.1.jar:/home/someuser/.m2/repository/org/jetbrains/kotlinx/kotlinx-coroutines-core-common/1.0.1/kotlinx-coroutines-core-common-1.0.1.jar:/home/someuser/.m2/repository/org/jetbrains/kotlinx/kotlinx-coroutines-reactive/1.0.1/kotlinx-coroutines-reactive-1.0.1.jar:/home/someuser/.m2/repository/org/reactivestreams/reactive-streams/1.0.2/reactive-streams-1.0.2.jar:/home/someuser/.m2/repository/com/graphql-java/graphql-java/11.0/graphql-java-11.0.jar:/home/someuser/.m2/repository/org/antlr/antlr4-runtime/4.7.1/antlr4-runtime-4.7.1.jar:/home/someuser/.m2/repository/com/graphql-java/java-dataloader/2.1.1/java-dataloader-2.1.1.jar:/home/someuser/.m2/repository/com/fasterxml/classmate/1.4.0/classmate-1.4.0.jar:/home/someuser/.m2/repository/com/fasterxml/jackson/core/jackson-core/2.9.7/jackson-core-2.9.7.jar:/home/someuser/.m2/repository/com/fasterxml/jackson/module/jackson-module-kotlin/2.9.7/jackson-module-kotlin-2.9.7.jar:/home/someuser/.m2/repository/com/fasterxml/jackson/core/jackson-annotations/2.9.0/jackson-annotations-2.9.0.jar:/home/someuser/.m2/repository/com/fasterxml/jackson/datatype/jackson-datatype-jdk8/2.9.7/jackson-datatype-jdk8-2.9.7.jar:/home/someuser/.m2/repository/com/esotericsoftware/reflectasm/1.11.7/reflectasm-1.11.7.jar:/home/someuser/.m2/repository/org/apache/commons/commons-lang3/3.8.1/commons-lang3-3.8.1.jar:/home/someuser/.m2/repository/com/google/guava/guava/26.0-jre/guava-26.0-jre.jar:/home/someuser/.m2/repository/com/google/code/findbugs/jsr305/3.0.2/jsr305-3.0.2.jar:/home/someuser/.m2/repository/org/checkerframework/checker-qual/2.5.2/checker-qual-2.5.2.jar:/home/someuser/.m2/repository/com/google/errorprone/error_prone_annotations/2.1.3/error_prone_annotations-2.1.3.jar:/home/someuser/.m2/repository/com/google/j2objc/j2objc-annotations/1.1/j2objc-annotations-1.1.jar:/home/someuser/.m2/repository/org/codehaus/mojo/animal-sniffer-annotations/1.14/animal-sniffer-annotations-1.14.jar:/home/someuser/.m2/repository/org/slf4j/slf4j-api/1.7.25/slf4j-api-1.7.25.jar com.coxautodev.graphql.tools.example.GraphqlJavaToolsExampleApplication

  .   ____          _            __ _ _
 /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
 \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
  '  |____| .__|_| |_|_| |_\__, | / / / /
 =========|_|==============|___/=/_/_/_/
 :: Spring Boot ::        (v2.1.0.RELEASE)

2019-03-25 16:15:26.781  INFO 29710 --- [           main] g.t.e.GraphqlJavaToolsExampleApplication : Starting GraphqlJavaToolsExampleApplication on datamaskinet with PID 29710 (/home/someuser/Documents/src/3rdParty/graphql-java-tools/example/target/classes started by someuser in /home/someuser/Documents/src/3rdParty/graphql-java-tools)
2019-03-25 16:15:26.786  INFO 29710 --- [           main] g.t.e.GraphqlJavaToolsExampleApplication : No active profile set, falling back to default profiles: default
2019-03-25 16:15:29.264  INFO 29710 --- [           main] org.xnio                                 : XNIO version 3.3.8.Final
2019-03-25 16:15:29.281  INFO 29710 --- [           main] org.xnio.nio                             : XNIO NIO Implementation Version 3.3.8.Final
2019-03-25 16:15:29.342  WARN 29710 --- [           main] io.undertow.websockets.jsr               : UT026009: XNIO worker was not set on WebSocketDeploymentInfo, the default worker will be used
2019-03-25 16:15:29.342  WARN 29710 --- [           main] io.undertow.websockets.jsr               : UT026010: Buffer pool was not set on WebSocketDeploymentInfo, the default pool will be used
2019-03-25 16:15:29.385  INFO 29710 --- [           main] io.undertow.servlet                      : Initializing Spring embedded WebApplicationContext
2019-03-25 16:15:29.386  INFO 29710 --- [           main] o.s.web.context.ContextLoader            : Root WebApplicationContext: initialization completed in 2521 ms
2019-03-25 16:15:29.484  INFO 29710 --- [           main] o.s.s.concurrent.ThreadPoolTaskExecutor  : Initializing ExecutorService 'applicationTaskExecutor'
2019-03-25 16:15:30.255  INFO 29710 --- [           main] o.s.b.a.e.web.EndpointLinksResolver      : Exposing 2 endpoint(s) beneath base path '/actuator'
2019-03-25 16:15:31.121  WARN 29710 --- [           main] c.c.graphql.tools.SchemaClassScanner     : Schema type was defined but can never be accessed, and can be safely deleted: PageInfo
2019-03-25 16:15:31.234  INFO 29710 --- [           main] c.o.m.g.b.GraphQLWebAutoConfiguration    : Building GraphQLObjectMapper including errorHandler: null
2019-03-25 16:15:31.244 ERROR 29710 --- [           main] c.o.m.g.b.e.GraphQLErrorHandlerFactory   : Cannot load class graphQLServletRegistrationBean. Error creating bean with name 'graphQLServletRegistrationBean': Requested bean is currently in creation: Is there an unresolvable circular reference?
2019-03-25 16:15:31.459  INFO 29710 --- [           main] o.s.b.w.servlet.FilterRegistrationBean   : Mapping filter: 'characterEncodingFilter' to: [/*]
2019-03-25 16:15:31.460  INFO 29710 --- [           main] o.s.b.w.servlet.FilterRegistrationBean   : Mapping filter: 'webMvcMetricsFilter' to: [/*]
2019-03-25 16:15:31.460  INFO 29710 --- [           main] o.s.b.w.servlet.FilterRegistrationBean   : Mapping filter: 'hiddenHttpMethodFilter' to: [/*]
2019-03-25 16:15:31.460  INFO 29710 --- [           main] o.s.b.w.servlet.FilterRegistrationBean   : Mapping filter: 'formContentFilter' to: [/*]
2019-03-25 16:15:31.460  INFO 29710 --- [           main] o.s.b.w.servlet.FilterRegistrationBean   : Mapping filter: 'requestContextFilter' to: [/*]
2019-03-25 16:15:31.460  INFO 29710 --- [           main] o.s.b.w.servlet.FilterRegistrationBean   : Mapping filter: 'httpTraceFilter' to: [/*]
2019-03-25 16:15:31.461  INFO 29710 --- [           main] o.s.b.w.servlet.FilterRegistrationBean   : Mapping filter: 'corsConfigurer' to: [/*]
2019-03-25 16:15:31.461  INFO 29710 --- [           main] o.s.b.w.servlet.ServletRegistrationBean  : Servlet dispatcherServlet mapped to [/]
2019-03-25 16:15:31.463  INFO 29710 --- [           main] o.s.b.w.servlet.ServletRegistrationBean  : Servlet configuredGraphQLHttpServlet mapped to [/graphql/*]
2019-03-25 16:15:31.515  INFO 29710 --- [           main] io.undertow.websockets.jsr               : UT026005: Adding programmatic server endpoint class graphql.servlet.GraphQLWebsocketServlet for path /subscriptions
2019-03-25 16:15:31.577  INFO 29710 --- [           main] o.s.b.w.e.u.UndertowServletWebServer     : Undertow started on port(s) 8080 (http) with context path ''
2019-03-25 16:15:31.579  INFO 29710 --- [           main] g.t.e.GraphqlJavaToolsExampleApplication : Started GraphqlJavaToolsExampleApplication in 5.252 seconds (JVM running for 5.975)
2019-03-25 16:15:32.099  INFO 29710 --- [on(2)-127.0.0.1] io.undertow.servlet                      : Initializing Spring DispatcherServlet 'dispatcherServlet'
2019-03-25 16:15:32.099  INFO 29710 --- [on(2)-127.0.0.1] o.s.web.servlet.DispatcherServlet        : Initializing Servlet 'dispatcherServlet'
2019-03-25 16:15:32.107  INFO 29710 --- [on(2)-127.0.0.1] o.s.web.servlet.DispatcherServlet        : Completed initialization in 8 ms

```

**Comment by gurgl** [comment]
Also, would be great for a newcomer with some directions on how to invoke a method.

**Comment by oliemansm** [comment]
This error is solved in graphql spring boot version 5.7.2 which will be release asap.
